File: Python3/553. Optimal Division.py
class Solution:
    def optimalDivision(self, nums):
        """
        :type nums: List[int]
        :rtype: str
        """
        nums = [str(num) for num in nums]
        if len(nums) == 1:
            return nums[0]
        if len(nums) == 2:
            return nums[0] + '/' + nums[1]
        return nums[0] + '/(' + '/'.join(nums[1:]) + ')'

File: Golang/553. Optimal Division.go
func optimalDivision(nums []int) string {
    s := make([]string, 0, len(nums))
    for _, num := range nums {
        s = append(s, strconv.Itoa(num))
    }
    if len(s) == 1 {
        return s[0]
    }
    if len(s) == 2 {
        return s[0] + "/" + s[1]
    }
    return s[0] + "/(" + strings.Join(s[1:], "/") + ")"
}
